sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -80,13 +80,49 @@ case class CreateTableLikeCommand(
         s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
     }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
 
-    catalog.createTable(tableToCreate, ifNotExists)
+    sourceTableDesc.tableType match {
+      case CatalogTableType.MANAGED | CatalogTableType.EXTERNAL | CatalogTableType.VIEW => // OK
+      case o => throw new AnalysisException(
+        s"CREATE TABLE LIKE is not allowed when the source table is ${o.name}")
+    }
+
+    if (DDLUtils.isDatasourceTable(sourceTableDesc) &&
+        sourceTableDesc.tableType != CatalogTableType.MANAGED) {
+      throw new AnalysisException(
+        "CREATE TABLE LIKE is not allowed when the source table is external tables created " +
+          "using the datasource API")
+    }
+
+    // For EXTERNAL_TABLE, the table properties has a particular field. To change it
+    // to a MANAGED_TABLE, we need to remove it; Otherwise, it will be EXTERNAL_TABLE,
+    // even if we set the tableType to MANAGED
+    // (metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1095-L1105)
+    // Table comment is stored as a table property. To clean it, we also should remove them.
+    val newTableProp =
+      sourceTableDesc.properties.filterKeys(key => key != "EXTERNAL" && key != "comment")
```

Comment:
That sounds true! See another fix by @yhuai in `toHiveTable`: https://github.com/apache/spark/blob/e679bc3c1cd418ef0025d2ecbc547c9660cac433/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L764-L770